test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1

```diff
@@ -67,15 +67,114 @@ Describe "Certificate Provider tests" -Tags "CI" {
         It "Should return PathNotFound when getting a non-existant certificate" {
             {Get-Item cert:\currentuser\my\IDONTEXIST -ErrorAction Stop} | Should -Throw -ErrorId "PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand"
         }
+        It "Should be able to get all service stores" {
+            $result = Get-Item -Path cert:\service
+            $result.LocationName | Should -Be Service
+        }
     }
     Context "Get-ChildItem tests"{
         It "should be able to get a container using a wildcard" {
             (Get-ChildItem Cert:\CurrentUser\M?).PSPath | Should -Be 'Microsoft.PowerShell.Security\Certificate::CurrentUser\My'
         }
-        It "Should return two items at the root of the provider" {
-            (Get-ChildItem -Path cert:\).Count | Should -Be 2
+        It "Should return three items at the root of the provider" {
+            (Get-ChildItem -Path cert:\).Count | Should -Be 3
+        }
+    }
+}
+
+Describe "Certificate service location tests" -Tags "CI", "RequireAdminOnWindows" {
+    BeforeAll{
+        if(!$IsWindows)
+        {
+            # Skip for non-Windows platforms
+            $defaultParamValues = $global:PSDefaultParameterValues.Clone()
+            $global:PSDefaultParameterValues = @{ "it:skip" = $true }
+            return
+        }
+
+        $serviceName = "pwsh-test-$([Guid]::NewGuid())"
+        $service = New-Service -Name $serviceName -BinaryPathName "$PSHome/pwsh.exe" -StartupType Manual
+    }
+
+    AfterAll {
+        if(!$IsWindows)
+        {
+            $global:PSDefaultParameterValues = $defaultParamValues
+            return
+        }
+
+        $service | Remove-Service
+    }
+
+    AfterEach {
+        if (!$IsWindows) {
+            return
+        }
+
+        $regPath = "HKLM:\SOFTWARE\Microsoft\Cryptography\Services\$serviceName"
+        if (Test-Path -LiteralPath $regPath) {
+            Remove-Item -LiteralPath $regPath -Recurse -Force
         }
     }
+
+    It "Errors with missing cert store" {
+        Get-Item Cert:\Service\$serviceName\My -ErrorAction SilentlyContinue -ErrorVariable err
+        $err.Count | Should -Be 1
+        [string]$err[0] | Should -Be "Cannot find path 'Cert:\Service\$serviceName\My' because it does not exist."
+    }
+
+    It "Creates cert store" {
+        $actual = New-Item Cert:\Service\$serviceName\Test
+        $actual | Should -BeOfType ([System.Security.Cryptography.X509Certificates.X509Store])
+        $actual.Name | Should -Be "$serviceName\Test"
+
+        (Get-Item Cert:\Service).StoreNames.Keys | Should -Contain "$serviceName\Test"
+    }
+
+    It "Get-Item on new cert store" {
+        New-Item Cert:\Service\$serviceName\Test | Out-Null
+
+        $actual = Get-Item Cert:\Service\$serviceName\Test
+        $actual | Should -BeOfType ([System.Security.Cryptography.X509Certificates.X509Store])
+        $actual.Name | Should -Be "$serviceName\Test"
+    }
+
+    It "Get-ChildItem on new cert store" {
```

Comment:
should there be additional validation to check for an error in the case `new-item` is called twice with the same name. Or whether `-force` in this case makes a difference?

Reply:
I would argue that's out of scope of this PR. `New-Item` can create stores in the `CurrentUser` or `LocalMachine` location so any existing behaviour will be inherited by `Service`. I did a brief test and there are a few things that could potentially change:

+ `-Force` does nothing if the store already exists so that should probably change
+ Currently they are thrown as exceptions, `New-Item` on a file is written as an error

So yes there probably should be some investigations into this but I don't think it should be bundled into this PR.